### PR TITLE
Bug 1686204: operator: periodically ensure default ingresscontroller

### PR DESCRIPTION
**Pull request by ironcladlou**
The default ingresscontroller is only ensured during operator startup,
necessitating an operator restart to restore defaults when the default
ingresscontroller is deleted.

This change is to periodically ensure the default ingresscontroller exists, so
defaults can be restored without affecting operator availability.

Fixes bug https://bugzilla.redhat.com/show_bug.cgi?id=1686204.

### CODE REVIEW

**Comment by ironcladlou** [comment]
The outcome of the default reconcile could potentially be reflected in operator status/conditions, but my immediate goal is to simply fix primary UX problem. Incorporating status will require more smarts/complexity.

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -201,15 +216,31 @@ func (o *Operator) Start(stop <-chan struct{}) error {
 	}
 }
 
-// ensureDefaultClusterIngress ensures that a default ClusterIngress exists.
-func (o *Operator) ensureDefaultClusterIngress() error {
-	ci := manifests.DefaultClusterIngress()
-	err := o.client.Create(context.TODO(), ci)
-	if err != nil && !errors.IsAlreadyExists(err) {
+// ensureDefaultClusterIngress creates the default ingresscontroller if it
```
Comment (Miciah) [inline pkg/operator/operator.go]: `ensureDefaultClusterIngress`→`ensureDefaultIngressController`

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -201,15 +216,31 @@ func (o *Operator) Start(stop <-chan struct{}) error {
...
-	} else if err == nil {
-		log.Info("created default clusteringress", "namespace", ci.Namespace, "name", ci.Name)
 	}
+	err = o.client.Create(context.TODO(), ic)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			return nil
+		} else {
+			return err
+		}
```
Comment (Miciah) [inline pkg/operator/operator.go]: Could simplify slightly:
```go
	if err := o.client.Create(context.TODO(), ic); err != nil {
		if errors.IsAlreadyExists(err) {
			return nil
		}
		return err
```

**Comment by danehans** [comment]
lgtm after addressing the two changes @Miciah suggested.

File: pkg/operator/operator.go (hunk 2 above)
Comment (ironcladlou) [reply pkg/operator/operator.go]: Fixed

File: pkg/operator/operator.go (hunk 1 above)
Comment (ironcladlou) [reply pkg/operator/operator.go]: Fixed

**Comment by ironcladlou** [comment]
All feedback addressed

**Pull request merged**